### ISSUE TITLE
Failed to bind to address http://localhost:49980.

### ISSUE THREAD

**Issue by Eilon** [aspnet/KestrelHttpServer#2277]
_From @livarcocc on January 24, 2018 17:9_

_From @Erave on January 24, 2018 11:43_

Yesterday, I have installed a major Windows Update and today I can't run my Asp.NET Core web app. I have a permission access problem dureing binding

Not existing process is listening on port 49980. I tried to stop firewall.
Do you see any reason to this exception ?

Thanks
Eric

```
System.IO.IOException: Failed to bind to address http://localhost:49980. ---> System.AggregateException: One or more errors occurred. (Error -4092 EACCES permission denied) (Error -4092 EACCES permission denied) ---> Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Networking.UvException: Error -4092 EACCES permission denied
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Networking.LibuvFunctions.ThrowError(Int32 statusCode)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Networking.LibuvFunctions.tcp_bind(UvTcpHandle handle, SockAddr& addr, Int32 flags)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Networking.UvTcpHandle.Bind(IPEndPoint endPoint)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Listener.ListenTcp(Boolean useFileHandle)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Listener.CreateListenSocket()
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Listener.<>c.<StartAsync>b__8_0(Listener listener)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.LibuvThread.CallbackAdapter`1.<>c.<.cctor>b__3_1(Object callback, Object state)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.LibuvThread.DoPostWork()






```

_Copied from original issue: dotnet/cli#8462_

_Copied from original issue: aspnet/Home#2821_

**Comment by Eilon** [comment]
_From @Tratcher on January 24, 2018 17:14_

Is it only that port? Does it work if you try a few random other ones?

**Comment by Eilon** [comment]
_From @muratg on January 24, 2018 17:52_

What's the output of `netstat -a -b` when you get this exception?

**Comment by sebastienros** [comment]
Got the same thing on linux

```
Unhandled Exception: System.IO.IOException: Failed to bind to address http://172.16.228.83:5000: address already in use. ---> Microsoft.AspNetCore.Protocols.AddressInUseException: Error -98 EADDRINUSE address already in use ---> Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Networking.UvException: Error -98 EADDRINUSE address already in use
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Networking.LibuvFunctions.ThrowError(Int32 statusCode)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Networking.UvTcpHandle.GetSockIPEndPoint()
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Listener.ListenTcp(Boolean useFileHandle)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Listener.<>c.<StartAsync>b__8_0(Listener listener)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.LibuvThread.DoPostWork()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.ListenerPrimary.StartAsync(String pipeName, Byte[] pipeMessage, IEndPointInformation endPointInformation, LibuvThread thread)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.LibuvTransport.BindAsync()
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.LibuvTransport.BindAsync()
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServer.<>c__DisplayClass22_0`1.<<StartAsync>g__OnBind|0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindEndpointAsync(ListenOptions endpoint, AddressBindContext context)
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindEndpointAsync(ListenOptions endpoint, AddressBindContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.ListenOptions.BindAsync(AddressBindContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.EndpointsStrategy.BindAsync(AddressBindContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindAsync(IServerAddressesFeature addresses, KestrelServerOptions serverOptions, ILogger logger, Func`2 createBinding)
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServer.StartAsync[TContext](IHttpApplication`1 application, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.StartAsync(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Hosting.WebHostExtensions.RunAsync(IWebHost host, CancellationToken token, String shutdownMessage)
   at Microsoft.AspNetCore.Hosting.WebHostExtensions.RunAsync(IWebHost host, CancellationToken token)
   at Microsoft.AspNetCore.Hosting.WebHostExtensions.Run(IWebHost host)
   at Benchmarks.Program.Main(String[] args) in /tmp/tmpQg2Oag.tmp/s3otnszp.iuq/benchmarks/src/Benchmarks/Program.cs:line 155
```

**Comment by muratg** [comment]
@sebastienros interesting... are you able to reproduce this on demand?  I'm assuming you're sure that there was nothing else occupying the port at the time?

**Comment by sebastienros** [comment]
After some investigations, it seems the `dotnet` process is still alive even after sending a `SIGINT` signal, even though the application has stopped.

The solution I am using is to wait for the process to be gone before starting a new application on the same port:

```
process.CloseMainWindow();

if (!process.HasExited)
{
    process.Kill();
}

process.Dispose();

do
{
    Log.WriteLine($"Waiting for process {processId} to stop ...");

    await Task.Delay(1000);

    try
    {
        process = Process.GetProcessById(processId);
    }
    catch
    {
        process = null;
    }

} while (process != null && !process.HasExited);

```

**Comment by aspnet-hello** [comment]
We periodically close 'discussion' issues that have not been updated in a long period of time.

We apologize if this causes any inconvenience. We ask that if you are still encountering an issue, please log a new issue with updated information and we will investigate.
